File: SS/SS-SPA/src/app/_models/artist.ts
import { SafeUrl } from '@angular/platform-browser';
import { Country } from './country';
import { UsState } from './usState';
import { UsCity } from './usCity';
import { UsZipCode } from './usZipCode';

export interface Artist {
  id: number;
  name: string;
  statusId: number;
  yearsActive: number;
  artistGroup: boolean;
  userId?: number;
  verified: boolean;
  usArtist: boolean;
  //
  currentLoc: ArtistLocation;
  //
  homeCountryId: number;
  homeCountry: Country;
  homeCity: string;
  homeUsStateId: number;
  homeUsState: UsState;
  homeUsCityId: number;
  homeUsCity: UsCity;
  homeUsZipCodeId: number;
  homeUsZipCode: UsZipCode;
  homeWorldRegionId: number;
  homeWorldCityId: number;
  currentCountryId: number;
  currentCityId: number;
  currentCity: string;
  mainPhotoURL: SafeUrl;
  careerBeginDate: Date;
  careerEndDate: Date;
  createdBy: number;
  createdDate: Date;
}

// export function initArtist(): Artist {
//   return {
//     id: -1,
//     name: null,
//   };
// }

export class Artist {
  public constructor(init?: Partial<Artist>) {
    Object.assign(this, init);
  }
}

export interface ArtistForm {
  name: string;
  usHomeCountry: boolean;
  homeCountryId: number;
  homeUsState: UsState;
  homeWorldRegion: any;
  homeUsCity: string;
  homeUsZipCode: string;
}

interface ArtistLocation {
  countryId: number;
  country: Country;
  city: string;
  usStateId: number;
  usState: UsState;
  usCityId: number;
  usCity: UsCity;
  usZipCodeId: number;
  usZipCode: UsZipCode;
}

File: SS/SS.API/SS/Data/Models/Artist.cs
using System;
using System.Collections.Generic;

namespace SS.Data.Models
{
    public partial class Artist
    {
        public Artist()
        {
            ArtistPhoto = new HashSet<ArtistPhoto>();
            ArtistTypeXref = new HashSet<ArtistTypeXref>();
        }

        public int ArtistId { get; set; }
        public string ArtistName { get; set; }
        public int? ArtistStatusId { get; set; }
        public DateTime CareerBeginDate { get; set; }
        public bool ArtistGroup { get; set; }
        public int? UserId { get; set; }
        public bool Verified { get; set; }
        public int HomeCountryId { get; set; }
        public int? HomeUscityId { get; set; }
        public int? HomeWorldCityId { get; set; }
        public int CurrentCountryId { get; set; }
        public int? CurrentUscityId { get; set; }
        public int? CurrentWorldCityId { get; set; }
        public int CreatedBy { get; set; }
        public DateTime CreatedDate { get; set; }

        public virtual ArtistStatus ArtistStatus { get; set; }
        public virtual Ssuser CreatedByNavigation { get; set; }
        public virtual Country CurrentCountry { get; set; }
        public virtual City CurrentUscity { get; set; }
        public virtual WorldCity CurrentWorldCity { get; set; }
        public virtual Country HomeCountry { get; set; }
        public virtual City HomeUscity { get; set; }
        public virtual WorldCity HomeWorldCity { get; set; }
        public virtual Ssuser User { get; set; }
        public virtual ICollection<ArtistPhoto> ArtistPhoto { get; set; }
        public virtual ICollection<ArtistTypeXref> ArtistTypeXref { get; set; }
    }
}
